### PR TITLE
RAXAJSS-261, RAXAJSS-262 RaxaAlert tables, Hibernate CRUD, Service Layers, DAO helper methods and respective tests created.

### PR DESCRIPTION
**Pull request by TarangMahajan**
Raxacore :
RaxaAlert Table created
RaxaAlert POJO created
RaxaAlert DAO created
RaxaAlert DAO implementation created
RaxaAlert Service created
RaxaAlert Service Implementation created
Tests created.

### CODE REVIEW

File: api/src/main/java/org/raxa/module/raxacore/RaxaAlert.java [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+	 * @should not get voided obs with three layers of hierarchy
+	 */
+	public Set<Obs> getObs() {
+		Set<Obs> ret = new HashSet<Obs>();
+		if (this.obs != null) {
+			for (Obs o : this.obs)
+				ret.addAll(getObsLeaves(o));
+			// this should be all thats needed unless the alert has been built by hand
+			//if (o.isVoided() == false && o.isObsGrouping() == false)
+			//      ret.add(o);
```
Comment (johnstoecker) [inline api/src/main/java/org/raxa/module/raxacore/RaxaAlert.java:228]: Tarang -- if the Obs has been voided by a doctor, I believe we don't want to include it. Why did you comment out these lines?

**Comment by johnstoecker** [comment]
Tarang, I can fix the Obs, so that's not a problem. However, I am getting an error when I add the updated code to OpenMRS as a .omod --- are you able to build + compile your code and add as a module to OpenMRS?

I am getting an error saying that 'liquibase.xml cannot be updated'

File: api/src/main/java/org/raxa/module/raxacore/RaxaAlert.java (hunk 1 above)
Comment (TarangMahajan) [inline api/src/main/java/org/raxa/module/raxacore/RaxaAlert.java:228]: Actually, the OpenMRS code didn't allow the doctor to void Obs, so I thought the same.

**Comment by TarangMahajan** [comment]
Well, I was able to build the omod and add before I added the raxaalert_providerrecipient table.

**Comment by johnstoecker** [comment]
Just confirmed that the new raxaalert_providerrecipient table is the problem -- please update this pull request without the joining table.

**Pull request merged**